public/app/features/alerting/unified/components/rule-editor/QueryEditor.tsx

```diff
@@ -1,224 +1,40 @@
 import { css } from '@emotion/css';
-import React, { PureComponent } from 'react';
+import React, { FC } from 'react';
 
-import {
```

Comment:
💬 This component has been cleaned up significantly as it was previously responsible for managing both data queries _and_ expression queries.

This logic has been moved to both the `QueriesAndExpressionsStep` and a separate `reducer.ts` to manage the data structures.